test/e2e_node/image_list.go

```diff
@@ -66,27 +63,6 @@ var NodePrePullImageList = sets.NewString(
 	"gcr.io/kubernetes-e2e-test-images/node-perf/tf-wide-deep-amd64:1.0",
 )
 
-// updateImageAllowList updates the framework.ImagePrePullList with
-// 1. the hard coded lists
-// 2. the ones passed in from framework.TestContext.ExtraEnvs
-// So this function needs to be called after the extra envs are applied.
-func updateImageAllowList() {
-	// Union NodePrePullImageList and PrePulledImages into the framework image pre-pull list.
-	framework.ImagePrePullList = NodePrePullImageList.Union(commontest.PrePulledImages)
-	// Images from extra envs
-	framework.ImagePrePullList.Insert(getNodeProblemDetectorImage())
-	framework.ImagePrePullList.Insert(getSRIOVDevicePluginImage())
-}
-
-func getNodeProblemDetectorImage() string {
```

Comment:
This function is used at another place as https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/97558/pull-kubernetes-verify/1346125853330771968

```
# k8s.io/kubernetes/test/e2e_node
vet: test/e2e_node/node_problem_detector_linux.go:66:11: undeclared name: getNodeProblemDetectorImage
make[1]: *** [Makefile:388: vet] Error 1
```

Reply:
@oomichi thanks for your review. 

I'm confusing about the analysis process of the `staticcheck` tool. 

because all the unused variables and unused functions in some go files are actually called by some test files.  it's confronted with compile-time errors when I remove these codes in order to fix it.

I retried several versions of the `staticcheck` tool but got the same result.

```shell
staticcheck -checks "all,-S1*,-ST1*" ./test/e2e_node ./test/e2e_node/builder ./test/e2e_node/environment ./test/e2e_node/perf/workloads ./test/e2e_node/perftype ./test/e2e_node/remote ./test/e2e_node/runner/local ./test/e2e_node/runner/remote ./test/e2e_node/services ./test/e2e_node/system

test/e2e_node/image_list.go:73:6: func updateImageAllowList is unused (U1000)
test/e2e_node/image_list.go:81:6: func getNodeProblemDetectorImage is unused (U1000)
test/e2e_node/image_list.go:235:6: func getSRIOVDevicePluginImage is unused (U1000)
test/e2e_node/util.go:62:5: var startServices is unused (U1000)
test/e2e_node/util.go:63:5: var stopServices is unused (U1000)
```
However，it will magically be fixed if an unused variable or unused function becomes exportable。